catroid/src/main/java/org/catrobat/catroid/content/actions/StoreCSVIntoUserListAction.kt

```diff
@@ -0,0 +1,117 @@
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.content.actions
+
+import android.util.Log
+import com.badlogic.gdx.scenes.scene2d.actions.TemporalAction
+import com.opencsv.CSVParserBuilder
+import com.opencsv.CSVReaderBuilder
+import com.opencsv.exceptions.CsvException
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.formulaeditor.InterpretationException
+import org.catrobat.catroid.formulaeditor.UserList
+import java.io.IOException
+import java.io.StringReader
+import java.util.regex.Pattern
+
+class StoreCSVIntoUserListAction : TemporalAction() {
+
+    private var sprite: Sprite? = null
+
+    private var formulaColumnToExtract: Formula? = null
+    private var formulaCSVData: Formula? = null
+
+    private var userList: UserList? = null
+
+    override fun update(percent: Float) {
+        userList ?: return
+
+        val csvData: String
+        try {
+            csvData = formulaCSVData?.interpretString(sprite) ?: ""
+
+            var columnToExtract: Int
+            columnToExtract = formulaColumnToExtract?.interpretInteger(sprite) ?: 0
+
+            columnToExtract--
+
+            var separator = ','
+            val pattern = Pattern.compile("^(?:\".*?\"|[^,;]*?)([,;])")
+            val matcher = pattern.matcher(csvData)
+            if (matcher.find()) {
+                separator = matcher.group(1)[0]
+            }
+            if (separator != ',' && separator != ';') {
+                separator = ','
+            }
+
+            userList?.reset()
+            var resetList = true
+
+            val parser = CSVParserBuilder().withSeparator(separator).build()
+            val reader = CSVReaderBuilder(StringReader(csvData)).withCSVParser(parser)
+                .build()
+
+            val allRows = reader.readAll()
+            if (allRows.size > 0 && columnToExtract >= 0) {
+                for (row in allRows) {
+                    if (row.size <= columnToExtract) {
+                        userList!!.addListItem("")
+                    } else {
+                        resetList = false
+                        userList!!.addListItem(row[columnToExtract])
+                    }
```

Comment:
please change in line 80&83 the userList!! to userList?
if you don't check for NPE don't enforce them, with the elvis-operator(?) kotlin ignores the function when it is NULL

Reply:
the rest of the conversion was fine